### DISCUSSION TITLE
Array as completions.parse's response_format

### DISCUSSION THREAD

**Discussion by CodyBontecou** [openai/openai-python#1734] [Q&A]
```python
class UserDetail(BaseModel):
    age: int
    name: str

response = client.beta.chat.completions.parse(
    model="gpt-4o-mini",
    messages=[
        {
            "role": "user",
            "content": "Provide synthetic data of 10 users.",
        }
    ],
    response_format=UserDetail,
)
```

I'd like to adjust the `response_format` to instead be an `Iterable[UserDetail]` and have the response return an array of UserDetail's rather than a single. 

Explicitly setting `response_format=Iterable[UserDetail]` gives me the error: `TypeError: issubclass() arg 1 must be a class`.

I was able to get this to work using [Instructor's](https://github.com/jxnl/instructor) implementation, so it's possible. But I'm not sure how they're making it work.

```python
client = instructor.from_openai(OpenAI())

response = client.chat.completions.create(
    model="gpt-4o-mini",
    response_model=Iterable[UserDetail],
    messages=[
        {
            "role": "user",
            "content": "Provide synthetic data of 10 users.",
        }
    ],
)
```

**Comment by issamhaimour** [comment]
> ```python
> 
> class UserDetail(BaseModel):
> 
>     age: int
> 
>     name: str
> 
> 
> 
> response = client.beta.chat.completions.parse(
> 
>     model="gpt-4o-mini",
> 
>     messages=[
> 
>         {
> 
>             "role": "user",
> 
>             "content": "Provide synthetic data of 10 users.",
> 
>         }
> 
>     ],
> 
>     response_format=UserDetail,
> 
> )
> 
> ```
> 
> 
> 
> I'd like to adjust the `response_format` to instead be an `Iterable[UserDetail]` and have the response return an array of UserDetail's rather than a single. 
> 
> 
> 
> Explicitly setting `response_format=Iterable[UserDetail]` gives me the error: `TypeError: issubclass() arg 1 must be a class`.
> 
> 
> 
> I was able to get this to work using [Instructor's](https://github.com/jxnl/instructor) implementation, so it's possible. But I'm not sure how they're making it work.
> 
> 
> 
> ```python
> 
> client = instructor.from_openai(OpenAI())
> 
> 
> 
> response = client.chat.completions.create(
> 
>     model="gpt-4o-mini",
> 
>     response_model=Iterable[UserDetail],
> 
>     messages=[
> 
>         {
> 
>             "role": "user",
> 
>             "content": "Provide synthetic data of 10 users.",
> 
>         }
> 
>     ],
> 
> )
> 
> ```

**Reply by CodyBontecou** [reply]
Am I missing something? I believe this is just a quote with no changes.
